mytoyota/models/endpoints/vehicle_guid.py

```diff
@@ -0,0 +1,391 @@
+    how_to_videos: bool = Field(alias="howToVideos")
+    hybrid_pulse: bool = Field(alias="hybridPulse")
+    hydrogen_pulse: bool = Field(alias="hydrogenPulse")
+    important_message: bool = Field(alias="importantMessage")
+    insurance: bool = Field(alias="insurance")
+    last_parked: bool = Field(alias="lastParked")
+    lcfs: bool = Field(alias="lcfs")
+    linked_accounts: bool = Field(alias="linkedAccounts")
+    maintenance_timeline: bool = Field(alias="maintenanceTimeline")
+    marketing_card: bool = Field(alias="marketingCard")
+    marketing_consent: bool = Field(alias="marketingConsent")
+    master_consent_editable: bool = Field(alias="masterConsentEditable")
+    my_destination: bool = Field(alias="myDestination")
+    owners_manual: bool = Field(alias="ownersManual")
+    paid_product: bool = Field(alias="paidProduct")
+    parked_vehicle_locator: bool = Field(alias="parkedVehicleLocator")
+    parking: bool = Field(alias="parking")
+    parking_notes: bool = Field(alias="parkingNotes")
+    personalized_settings: bool = Field(alias="personalizedSettings")
+    privacy: bool = Field(alias="privacy")
+    recent_trip: bool = Field(alias="recentTrip")
+    remote_dtc: bool = Field(alias="remoteDtc")
+    remote_parking: bool = Field(alias="remoteParking")
+    remote_service: bool = Field(alias="remoteService")
+    roadside_assistance: bool = Field(alias="roadsideAssistance")
+    safety_recall: bool = Field(alias="safetyRecall")
+    schedule_maintenance: bool = Field(alias="scheduleMaintenance")
+    service_history: bool = Field(alias="serviceHistory")
+    shop_genuine_parts: bool = Field(alias="shopGenuineParts")
+    smart_charging: bool = Field(alias="smartCharging")
+    ssa_download: bool = Field(alias="ssaDownload")
+    sxm_radio: bool = Field(alias="sxmRadio")
+    telemetry: bool = Field(alias="telemetry")
+    tff: bool = Field(alias="tff")
+    tire_pressure: bool = Field(alias="tirePressure")
+    v1g: bool = Field(alias="v1g")
+    va_setting: bool = Field(alias="vaSetting")
+    vehicle_diagnostic: bool = Field(alias="vehicleDiagnostic")
+    vehicle_health_report: bool = Field(alias="vehicleHealthReport")
+    vehicle_specifications: bool = Field(alias="vehicleSpecifications")
+    vehicle_status: bool = Field(alias="vehicleStatus")
+    we_hybrid: bool = Field(alias="weHybrid")
+    wifi: bool = Field(alias="wifi")
+    xcapp: bool = Field(alias="xcapp")
+
+
+class VehicleGuidModel(BaseModel):
+    alerts: List[Any]  # TODO unsure what this returns
+    asiCode: str
+    brand: str
+    capabilities: List[_CapabilitiesModel]
+    car_line_name: str = Field(alias="carlineName")
+    color: str
+    commercial_rental: bool = Field(alias="commercialRental")
+    contract_id: str = Field(alias="contractId")
+    cts_links: _LinksModel = Field(alias="ctsLinks")
+    data_consent: _DataConsentModel = Field(alias="dataConsent")
+    date_of_first_use: date = Field(alias="dateOfFirstUse")
```

Comment:
```suggestion
    date_of_first_use: Optional[date] = Field(alias="dateOfFirstUse")
```

My car is returning `dateOfFirstUse` as `None` which causes a validation error:

```
pydantic_core._pydantic_core.ValidationError: 1 validation error for VehiclesResponseModel
payload.0.dateOfFirstUse
  Input should be a valid date [type=date_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.5/v/date_type
```

It is probably related to being brand new, I've just got it so I don't know when or how they fill this field but it is returning `None` so I guess it should be considered.

Reply:
Good catch 👍🏻